src/securityPolicy.c

```diff
@@ -293,21 +302,39 @@ security_policy_t policyForSignTxOutputToken(
         token_amount_t* token MARK_UNUSED
 )
 {
-	ALLOW_IF(addressPolicy == POLICY_ALLOW_WITHOUT_PROMPT);
+	switch (addressPolicy) {
+	case POLICY_ALLOW_WITHOUT_PROMPT:
+		ALLOW();
 
-	SHOW();
+	case POLICY_SHOW_BEFORE_RESPONSE:
+		SHOW();
+
+	default:
+		ASSERT(false);
+	}
+
+	DENY(); // should not be reached
 }
 
 security_policy_t policyForSignTxOutputConfirm(
```

Comment:
Is there any point in separating the policies for different output entries if they are basically the same? I can see them getting unintentionally misaligned if we opt to change outputs and general and forget about tokens or vice-versa.

So I would opt for a single "policyForSignTxOutputEntries" which seems like what we indeed want to achieve there and split it only once we would actually need to.